benchmark/assert/deepequal-buffer.js

```diff
@@ -1,39 +1,62 @@
 'use strict';
 const common = require('../common.js');
 const assert = require('assert');
+
 const bench = common.createBenchmark(main, {
-  n: [1e3],
+  n: [1e5],
```

Comment:
I'd be tempted to suggest adding as an extra - unless something has been found to be not valid with it being 1e3. 
@joyeecheung implemented this benchmark in the first place, was there any particular reason for choosing 1e3 and not a larger number?
The other thing to consider is the length of time to run the benchmark, we should be careful about adding more variants which would add to longer run times for the benchmark suite